### PR TITLE
[README] Specify in more detail how to create the development `datadog.yml`

### LINKED ISSUE #14295: [BUG] line 1: cannot unmarshal !!map into string
I've built datadog-agent from source using the provided instructions.

I used the example datadog.yaml script provided in https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml

When I run the agent, I get:

`Error: unable to load Datadog config file: While parsing config: yaml: unmarshal errors:
  line 1: cannot unmarshal !!map into string`

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?

Specify in more detail how to create the development `datadog.yml`.

### Motivation

Build instructions old wording made it ambiguous on how to create the development `datadog.yml`. This change should hopefully make things a bit clearer.

### Additional Notes

Fixes: https://github.com/DataDog/datadog-agent/issues/14295

### Possible Drawbacks / Trade-offs

N/A

### Describe how to test/QA your changes

N/A

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ To start working on the Agent, you can build the `main` branch:
 1. Checkout the repo: `git clone https://github.com/DataDog/datadog-agent.git $GOPATH/src/github.com/DataDog/datadog-agent`.
 2. cd into the project folder: `cd $GOPATH/src/github.com/DataDog/datadog-agent`.
 3. Install go tools: `invoke install-tools`.
-4. Create a development `datadog.yaml` configuration file in `dev/dist/datadog.yaml`, containing a valid API key: `api_key: <API_KEY>`
+4. Create a development `datadog.yaml` configuration file in `dev/dist/datadog.yaml`, containing a valid API key: `api_key: <API_KEY>`. You can either start with an empty one or use the full one generated by the Agent build from step 5 (located in `cmd/agent/dist/datadog.yaml` after the build finishes).
```
Comment (alai97) [inline README.md:48]:
```suggestion
4. Create a development `datadog.yaml` configuration file in `dev/dist/datadog.yaml`, containing a valid API key: `api_key: <API_KEY>`. You can either start with an empty one or use the full one generated by the Agent build from Step 5 (located in `cmd/agent/dist/datadog.yaml` after the build finishes).
```

**Review by alai97 — APPROVED** [review]
Looks good for docs!

**Review by mx-psi — APPROVED** [review]
TIL :open_mouth:

**Review by hush-hush — APPROVED** [review]

**Comment by sgnn7** [comment]
Failing windows test is not related. Merging.

**Pull request merged**